### ISSUE TITLE
3.0: Upgrade Screen is Confusing

### ISSUE THREAD

**Issue by spencerfinnell** [easydigitaldownloads/easy-digital-downloads#6880]
I've tested 3.0 on a couple of sites and using the upgrade screen has been a bit confusing to me during all experiences.

- When I click "Upgrade Database" on the first step it triggers all of the steps automatically. If they happen all at once why not make one button initially?
- The "Upgrades" page title is pretty generic. I think "Easy Digital Downloads 3.0 Upgrades" would give more context.
- Not a huge fan of the mdash: https://cloudup.com/cNqOeq_GbuQ Either have it on a new line without the dash or remove it completely. It just duplicates what the success alert below displays.
- Same thing with this state: https://cloudup.com/cO3ydsEbGRb -- I think it would be better to use a warning similar to the alert in the previous note about how migration is no longer possible.

Functionality wise:

- When I refresh I can still upgrade orders, order notes, and customer notes again. There are no errors so I'm not sure if it's failing or not. I think it's just not auto removing the legacy data and customer addresses are? 

---

I think only surfacing a single step for the upgrades will be a lot easier for customers. In the end they really don't care what is going on as long as things don't break. 

If something fails they shouldn't be using their site and should contact support. If everything goes according to plan they should just continue their day as normal; which means they probably also shouldn't have to think about legacy data. 

Anyone who cares about legacy data after a successful upgrade knows what they are doing and has made appropriate plans for it already. What about setting a future (like 6 months) to delete that data automatically? Provides a pretty long safety net for catching edge cases and keeps the user as a user not a database admin.

---

Proposal:

Simplify the upgrade screen: https://cloudup.com/cMWAvdHbeAH

Perform all the upgrades at once and schedule a cron to remove legacy data in the very far future (and create a tool that users can trigger manually). If something goes wrong display an error and a very clear way to create a support ticket with relevant information.

---

As someone who experienced the Ninja Forms 3.0 update on many sites (and dealt with many customers using the plugin) I've realized how "unimportant" these kind of upgrades really are and how behind the scenes they should be kept.  Obviously the user needs to trigger it so backups (lol) can be made but other than that it should be irrelevant to them.

**Comment by cklosowski** [comment]
Actually the first step is required to go first, it's the initial migration, if it doesn't migrate all items it should not auto start the next steps.

We used this same method for software licensing so people knew what was happening instead of blindly failing at some point. Failing on a specific step helps us debug better.

**Comment by spencerfinnell** [comment]
But code can make the first step go first (and fix #6810). Definitely tough knowing exactly where things failed but maybe an upgrade-specific system info could be generated to help give support a hint at what's going wrong.

I just think putting the weight of knowing or caring that there is a separate order notes database on the user is unnecessary.

**Comment by sunnyratilal** [comment]
One way we could go about doing this is having one big upgrade and having a list of the upgrades and a tick that just appears next to them after the upgrade is complete.

If there's an error, we display a warning icon with a Retry button?

**Comment by spencerfinnell** [comment]
Seems like a good middle ground. We do that for our content importer and it does provide a way for people to screenshot where things went wrong. 

Shows them things are happening without requiring interaction.

**Comment by cklosowski** [comment]
I do think we should still expose the steps, just maybe not in their own metaboxes. This will help us identify sections that fail easier. I'll see If I can tighetn up the UI a bit.

**Comment by nosegraze** [comment]
I just ran the upgrade on a test site and experienced one possible confusing point.

I clicked "Upgrade" on step 1, which automatically processed everything up through "Step 3: Upgrade Customer Email Addresses". But then the auto upgrade stopped on the next step and showed this message:

```
Step 4: Upgrade Discounts
Legacy data has already been removed, migration is not possible at this time.
```

I had to manually trigger step 5 to keep going.

I think this is potentially unclear because my screen size only showed through step 3 and it looked like it was doing all the steps automatically. If someone didn't scroll down they may not have realized that the upgrade actually stopped and they had to manually continue it.
